<a href="https://colab.research.google.com/github/Aggie2024/Classification_Projects/blob/main/Classification%20Project%20Datasets.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
# Importing required packages
import pandas as pd

# ADULT INCOME DATASET

In [2]:
# Save the filepath as a variable "fpath"
fpath =  "/content/drive/MyDrive/CodingDojo/Project_AdultIncome/adult.csv"
# Read in the file and display the data
df_income = pd.read_csv(fpath)
# preview of the data
df_income.head()

age  workclass  fnlwgt     education  educational-num      marital-status  \
0   25    Private  226802          11th                7       Never-married   
1   38    Private   89814       HS-grad                9  Married-civ-spouse   
2   28  Local-gov  336951    Assoc-acdm               12  Married-civ-spouse   
3   44    Private  160323  Some-college               10  Married-civ-spouse   
4   18          ?  103497  Some-college               10       Never-married   

          occupation relationship   race  gender  capital-gain  capital-loss  \
0  Machine-op-inspct    Own-child  Black    Male             0             0   
1    Farming-fishing      Husband  White    Male             0             0   
2    Protective-serv      Husband  White    Male             0             0   
3  Machine-op-inspct      Husband  Black    Male          7688             0   
4                  ?    Own-child  White  Female             0             0   

   hours-per-week native-country income  
0              40  United-States  <=50K  
1              50  United-States  <=50K  
2              40  United-States   >50K  
3              40  United-States   >50K  
4              30  United-States  <=50K

In [3]:
# summary of the columns
df_income.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48842 entries, 0 to 48841
Data columns (total 15 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   age              48842 non-null  int64 
 1   workclass        48842 non-null  object
 2   fnlwgt           48842 non-null  int64 
 3   education        48842 non-null  object
 4   educational-num  48842 non-null  int64 
 5   marital-status   48842 non-null  object
 6   occupation       48842 non-null  object
 7   relationship     48842 non-null  object
 8   race             48842 non-null  object
 9   gender           48842 non-null  object
 10  capital-gain     48842 non-null  int64 
 11  capital-loss     48842 non-null  int64 
 12  hours-per-week   48842 non-null  int64 
 13  native-country   48842 non-null  object
 14  income           48842 non-null  object
dtypes: int64(6), object(9)
memory usage: 5.6+ MB


1. Source of data
- The data comes from The Census Income Data Set from the UCI Machine Learning Repository named as Adult income or adult dataset.
https://archive.ics.uci.edu/dataset/2/adult

2. Brief description of data
- An individual’s annual income results from various factors. Intuitively, it is influenced by the individual’s education level, age, gender, occupation, and etc.

3. What is the target?
- The purpose of classification is to predict, whether an income exceeds 50k per year.

4. What does one row represent?
- Each row represents an individual.

5. Is this a classification or regression problem?
- This is a classification problem.

6. How many features does the data have?
- Data has 15 features.

7. How many rows are in the dataset?
- Data has 48842 rows.

8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?
- From a glance, the data has enough data points to do a good job at modeling.

**Features in this dataset**

|Column Name | Description |
|----------- | ----------- |
|age| continuous|
|workclass| Private, Self-emp-not-inc, Self-emp-inc, Federal-gov, Local-gov, State-gov, Without-pay, Never-worked|
|fnlwgt| continuous|
|education| Bachelors, Some-college, 11th, HS-grad, Prof-school, Assoc-acdm, Assoc-voc, 9th, 7th-8th, 12th, Masters, 1st-4th, 10th, Doctorate, 5th-6th, Preschool|
|education-num | continuous|
|marital-status| Married-civ-spouse, Divorced, Never-married, Separated, Widowed, Married-spouse-absent, Married-AF-spouse|
|occupation| Tech-support, Craft-repair, Other-service, Sales, Exec-managerial, Prof-specialty, Handlers-cleaners, Machine-op-inspct, Adm-clerical, Farming-fishing, Transport-moving, Priv-house-serv, Protective-serv, Armed-Forces|
|relationship| Wife, Own-child, Husband, Not-in-family, Other-relative, Unmarried|
|race| White, Asian-Pac-Islander, Amer-Indian-Eskimo, Other, Black|
|sex| Female, Male|
|capital-gain| continuous|
|capital-loss| continuous|
|hours-per-week| continuous|
|native-country| United-States, Cambodia, England, Puerto-Rico, Canada, Germany, Outlying-US(Guam-USVI-etc), India, Japan, Greece, South, China, Cuba, Iran, Honduras, Philippines, Italy, Poland, Jamaica, Vietnam, Mexico, Portugal, Ireland, France, Dominican-Republic, Laos, Ecuador, Taiwan, Haiti, Columbia, Hungary, Guatemala, Nicaragua, Scotland, Thailand, Yugoslavia, El-Salvador, Trinadad&Tobago, Peru, Hong, Holand-Netherlands|
class: >50K, <=50K

# WATER PUMPS DATASET

In [4]:
# Save the filepath as a variable "fpath"
file_path =  "/content/drive/MyDrive/CodingDojo/Project_WaterPump/training_Set_values.csv"
# Read in the file and display the data
df_pump = pd.read_csv(file_path)
# preview of the data
df_pump.head()

id  amount_tsh date_recorded        funder  gps_height     installer  \
0  69572      6000.0    2011-03-14         Roman        1390         Roman   
1   8776         0.0    2013-03-06       Grumeti        1399       GRUMETI   
2  34310        25.0    2013-02-25  Lottery Club         686  World vision   
3  67743         0.0    2013-01-28        Unicef         263        UNICEF   
4  19728         0.0    2011-07-13   Action In A           0       Artisan   

   longitude   latitude              wpt_name  num_private  ... payment_type  \
0  34.938093  -9.856322                  none            0  ...     annually   
1  34.698766  -2.147466              Zahanati            0  ...    never pay   
2  37.460664  -3.821329           Kwa Mahundi            0  ...   per bucket   
3  38.486161 -11.155298  Zahanati Ya Nanyumbu            0  ...    never pay   
4  31.130847  -1.825359               Shuleni            0  ...    never pay   

  water_quality quality_group      quantity  quantity_group  \
0          soft          good        enough          enough   
1          soft          good  insufficient    insufficient   
2          soft          good        enough          enough   
3          soft          good           dry             dry   
4          soft          good      seasonal        seasonal   

                 source           source_type  source_class  \
0                spring                spring   groundwater   
1  rainwater harvesting  rainwater harvesting       surface   
2                   dam                   dam       surface   
3           machine dbh              borehole   groundwater   
4  rainwater harvesting  rainwater harvesting       surface   

               waterpoint_type waterpoint_type_group  
0           communal standpipe    communal standpipe  
1           communal standpipe    communal standpipe  
2  communal standpipe multiple    communal standpipe  
3  communal standpipe multiple    communal standpipe  
4           communal standpipe    communal standpipe  

[5 rows x 40 columns]

In [5]:
# summary of the columns
df_pump.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 59400 entries, 0 to 59399
Data columns (total 40 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   id                     59400 non-null  int64  
 1   amount_tsh             59400 non-null  float64
 2   date_recorded          59400 non-null  object 
 3   funder                 55763 non-null  object 
 4   gps_height             59400 non-null  int64  
 5   installer              55745 non-null  object 
 6   longitude              59400 non-null  float64
 7   latitude               59400 non-null  float64
 8   wpt_name               59398 non-null  object 
 9   num_private            59400 non-null  int64  
 10  basin                  59400 non-null  object 
 11  subvillage             59029 non-null  object 
 12  region                 59400 non-null  object 
 13  region_code            59400 non-null  int64  
 14  district_code          59400 non-null  int64  
 15  lg

1. Source of data
- Taarifa is an open source platform for the crowd sourced reporting and triaging of infrastructure related issues. Think of it as a bug tracker for the real world which helps to engage citizens with their local government. We are currently working on an Innovation Project in Tanzania, with various partners.
The data comes from the Taarifa waterpoints dashboard, which aggregates data from the Tanzania Ministry of Water.

2. Brief description of data
- The data is from Taarifa and the Tanzanian Ministry of Water with a number of variables about what kind of pump is operating, when it was installed, and how it is managed etc. The variables can be used to predict 3 classes: which pumps are functional, which need some repairs, and which don't work at all. An understanding of which waterpoints will fail can improve maintenance operations and ensure that clean, potable water is available to communities across Tanzania.

3. What is the target?
- The task is to use data from Taarifa and the Tanzanian Ministry of Water to predict the operating condition of a waterpoint for each record in the dataset ie which pumps are functional, which need some repairs, and which don't work at all.

4. What does one row represent?
- One row represents one of Taarifa's water pumps.

5. Is this a classification or regression problem?
- This is a classification problem.

6. How many features does the data have?
- There are 40 features in the data set.

7. How many rows are in the dataset?
- There are 59400 rows in the dataset.

8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?
- Due to the number of features and missing values, the data will take time for cleaning. The data set is also already split into numerous data sets.

The features in this dataset

|Column Name | Description |
|----------- | ----------- |
|amount_tsh | Total static head (amount water available to waterpoint)|
|date_recorded|  The date the row was entered|
|funder| Who funded the well|
|gps_height| Altitude of the well|
|installer | Organization that installed the well|
|longitude | GPS coordinate|
|latitude | GPS coordinate|
|wpt_name | Name of the waterpoint if there is one|
|num_private|   |
|basin | Geographic water basin|
|subvillage | Geographic location|
|region| Geographic location|
|region_code| Geographic location (coded)|
|district_code| Geographic location (coded)|
|lga| Geographic location|
|ward| Geographic location|
|population| Population around the well|
|public_meeting| True/False|
|recorded_by | Group entering this row of data|
|scheme_management | Who operates the waterpoint|
|scheme_name | Who operates the waterpoint|
|permit | If the waterpoint is permitted|
|construction_year | Year the waterpoint was constructed|
|extraction_type | The kind of extraction the waterpoint uses|
|extraction_type_group | The kind of extraction the waterpoint uses|
|extraction_type_class | The kind of extraction the waterpoint uses|
|management | How the waterpoint is managed|
|management_group | How the waterpoint is managed|
|payment | What the water costs|
|payment_type | What the water costs|
|water_quality | The quality of the water|
|quality_group | The quality of the water|
|quantity | The quantity of water|
|quantity_group | The quantity of water|
|source | The source of the water|
|source_type | The source of the water|
|source_class | The source of the water|
|waterpoint_type | The kind of waterpoint|
|waterpoint_type_group | The kind of waterpoint|



In [6]:
# Save the filepath as a variable "fpath"
file_path1 =  "/content/drive/MyDrive/CodingDojo/Project_WaterPump/training_Set_labels.csv"
# Read in the file and display the data
df_pump1 = pd.read_csv(file_path1)
# preview of the data
df_pump1.info()
df_pump1.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 59400 entries, 0 to 59399
Data columns (total 2 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   id            59400 non-null  int64 
 1   status_group  59400 non-null  object
dtypes: int64(1), object(1)
memory usage: 928.2+ KB


id    status_group
0  69572      functional
1   8776      functional
2  34310      functional
3  67743  non functional
4  19728      functional

**Distribution of Labels**

The labels in this dataset are simple. There are three possible values:

- functional - the waterpoint is operational and there are no repairs needed
- functional needs repair - the waterpoint is operational, but needs repairs
- non functional - the waterpoint is not operational
